libs/api/libbpf_program.cpp

```diff
@@ -235,6 +235,27 @@ bpf_prog_attach(int prog_fd, int attachable_fd, enum bpf_attach_type type, unsig
     return 0;
 }
 
+int
+bpf_prog_detach2(int prog_fd, int attachable_fd, enum bpf_attach_type type)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+    if (_does_attach_type_support_attachable_fd(type)) {
+        result = ebpf_program_detach(prog_fd, get_ebpf_attach_type(type), &attachable_fd, sizeof(attachable_fd));
```

Comment:
As noted in comments in ebpf_api.cpp, currently ebpf_program_detach can throw an exception on out of memory, but bpf_prog_detach2() is not supposed to throw an exception.